### PR TITLE
Auth2

### PR DESCRIPTION
**Pull request by malamili**
I would say a pre-publish super alpha functionality
- UserService needs refactoring, so please ignore it for now during the review
- Still some tests need to be made
- WeaverProject and WeaverACL still need work

### CODE REVIEW

File: config/default.coffee [hunk 1]
```diff
@@ -1,9 +1,28 @@
...
+        endpoint: 'http://localhost:9000'
+        accessKey: 'NYLEXGR6MF2IE99LZ4UE'
+        secretKey: 'CjMuTRwYPcneXnGac2aQH0J+EdYdehTW4Cw7bZGD'
+    }
+    {
+      database: 'http://localhost:9475'
+      fileServer:
+        endpoint: 'http://localhost:9000'
+        accessKey: 'NYLEXGR6MF2IE99LZ4UE'
+        secretKey: 'CjMuTRwYPcneXnGac2aQH0J+EdYdehTW4Cw7bZGD'
```
Comment (carvilsi) [inline config/default.coffee]: I' missing the 'secure' and 'region' keys

File: src/auth/AuthCtrl.coffee [hunk 2]
```diff
@@ -1,185 +1,55 @@
...
-    doLogInCall(res, 'token',req.payload.user,req.payload.password)
-    .then((re) ->
-      Promise.resolve(re)
-    ).catch((err) ->
-      errorCodeParserFlock(err)
-    )
+# Sign up a new user.
+bus.public("auth.signUp").require('userId', 'username', 'password', 'email').on((req, userId, username, password, email)->
+  authToken = UserService.signUp(userId, username, email, password)
+  authToken
```
Comment (carvilsi) [inline src/auth/AuthCtrl.coffee]: Why not just:
`bus.public("auth.signUp").require('userId', 'username', 'password', 'email').on((req, userId, username, password, email)->
UserService.signUp(userId, username, email, password)`

File: src/auth/AuthCtrl.coffee [hunk 3]
```diff
@@ -1,185 +1,55 @@
...
 )
 
+# Sign in existing user
+bus.public("auth.signIn").require('username', 'password').on((req, username, password) ->
+  authToken = UserService.signIn(username, password)
+  authToken
+)
```
Comment (carvilsi) [inline src/auth/AuthCtrl.coffee]: Same as above.

File: config/default.coffee (hunk 1 above)
Comment (malamili) [inline config/default.coffee]: yes, these seemed to be always the same (us-west, secure false) so i didnt see the need of putting them in config

File: src/auth/AuthCtrl.coffee (hunk 2 above)
Comment (malamili) [inline src/auth/AuthCtrl.coffee]: correct..i think this is a remainder of some console logging in the middle..will clean it ;)

File: src/auth/AuthCtrl.coffee (hunk 3 above)
Comment (malamili) [inline src/auth/AuthCtrl.coffee]: 👍

File: config/default.coffee (hunk 1 above)
Comment (carvilsi) [inline config/default.coffee]: I think that at least the secure must be at the init call for Minio (via config or hardcoded): https://docs.minio.io/docs/javascript-client-quickstart-guide

File: src/core/EventListener.coffee [hunk 4]
```diff
@@ -20,35 +21,49 @@ class EventListener
...
       catch error
         isErrorObject = Object.prototype.toString.call(error) is '[object Error]'
 
         # TODO: Make all errors error objects, or log the error here
         if isErrorObject
-          reject({code: -1, message: error.message})
+          return Promise.reject({code: -1, message: error.message})
         else
-          reject(error)
+          return Promise.reject(error)
```
Comment (gvdent) [inline src/core/EventListener.coffee]: I'm now convinced I missed the explicit return style decree :)

File: src/database/WeaverQueryCtrl.coffee [hunk 5]
```diff
@@ -1,28 +1,14 @@
...
-      new DatabaseService(endpoint)
-    )
+  database.query(req.payload.query).then((results) ->
 
-
-bus.private('query').require('target').on((req, target) ->
-  getDb(target).then((db) ->
-    db.query(req.payload.query)
+    #TODO Check permission per node before returning results
+    results
```
Comment (gvdent) [inline src/database/WeaverQueryCtrl.coffee]: or didn't I? :D

Comment (malamili) [inline src/database/WeaverQueryCtrl.coffee]: haha no i wasnt consistent..fixing it

**Pull request merged**